transport/http/src/test/java/org/kaazing/gateway/transport/http/specification/httpxe/OriginSecurityIT.java

```diff
@@ -0,0 +1,136 @@
+import org.kaazing.gateway.transport.http.HttpAcceptSession;
+import org.kaazing.gateway.transport.http.HttpAcceptorRule;
+import org.kaazing.gateway.transport.http.HttpStatus;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.kaazing.test.util.ITUtil.createRuleChain;
+
+public class OriginSecurityIT {
+
+    private static final ResourceAddress HTTP_ADDRESS = httpAddress();
+
+    private final K3poRule k3po = new K3poRule().setScriptRoot("org/kaazing/specification/httpxe/origin");
+
+    private final HttpAcceptorRule acceptor = new HttpAcceptorRule();
+
+    @Rule
+    public final TestRule chain = createRuleChain(acceptor, k3po);
+
+    @Test
+    @Specification("request.with.origin.header/request")
+    public void shouldPassWithOriginRequestHeader() throws Exception {
+        test(HTTP_ADDRESS, "http://source.example.com:80");
+    }
+
+    @Test
+    @Specification("bad.request.with.origin.header/request")
+    public void shouldFailWithOriginRequestHeader() throws Exception {
+        test(HTTP_ADDRESS, "http://unknownsource.example.com:80");
+    }
+
+    @Test
+    @Specification("request.with.origin.header.and.x.origin.header/request")
+    public void shouldPassWithOriginAndXoriginRequests() throws Exception {
+        test(HTTP_ADDRESS, "http://source.example.com:80");
+    }
+
+    @Test
+    @Specification("bad.request.with.origin.header.and.x.origin.header/request")
+    public void shouldFailWithOriginAndXoriginRequests() throws Exception {
+        test(HTTP_ADDRESS, "http://source.example.com:80");
+    }
+
+    @Test
+    @Specification("origin.request.using.ko.parameter/request")
+    public void shouldPassWhenUsingKoParameter() throws Exception {
+        test(HTTP_ADDRESS, "http://source.example.com:80");
+    }
+
+    @Test
+    @Specification("origin.request.using.referer/request")
+    public void shouldPassWithOnlyRefererAndXoriginRequest() throws Exception {
+        test(HTTP_ADDRESS, "http://source.example.com:80");
+    }
+
+    @Test
+    @Specification("x.origin.header.not.identical.to.origin.header/request")
+    public void shouldPassWhenXoriginHeaderDiffersFromOriginHeader() throws Exception {
+        test(HTTP_ADDRESS, "http://source.example.com:80");
+    }
+
+    @Test
+    @Specification("request.with.kac.parameter/request")
+    public void shouldPassWithAccessControlWithKacParameter() throws Exception {
+        test(HTTP_ADDRESS, "http://localhost:8000");
+    }
+
+    @Test
+    @Specification("x.origin.encoded.request.header/request")
+    public void shouldPassWithEncodedXoriginRequest() throws Exception {
+        test(HTTP_ADDRESS, "http://source.example.com:80");
+    }
+
+    private void test(ResourceAddress address, String origin) throws Exception {
```

Comment:
You will probably need a variant of this for the failure cases. Consider renaming to shouldPass (then the other can be shouldFail)